compiler-core/src/erlang/tests.rs

```diff
@@ -658,3 +658,163 @@ pub fn main() {
 "
     );
 }
+
+// https://github.com/gleam-lang/gleam/issues/3382
+#[test]
+fn function_named_module_info() {
+    assert_erl!(
+        "
+pub type ModuleInfo {
+    ModuleInfo
+}
+
+pub fn module_info() {
+    ModuleInfo
+}
+
+pub const constant = module_info
+
+pub fn main() {
+    module_info()
+
+    let function = module_info
+    function()
+
+    constant()
+
+    function
+}
```

Comment:
One thing being tested per test please! Split this into a test for each way that the function is being rendered